src/inmanta/server/services/policy_engine_service.py

```diff
@@ -0,0 +1,199 @@
+            raise Exception("Policy engine is not running. Call OpaServer.start() first.")
+        client = SimpleAsyncHTTPClient()
+        policy_engine_addr = await self._opa_process.get_addr_policy_engine()
+        request = HTTPRequest(
+            url=f"http://{policy_engine_addr}/v1/data/policy/allowed",
+            method="POST",
+            headers={"Content-Type": "application/json"},
+            body=json.dumps(input_data, default=util.api_boundary_json_encoder),
+        )
+        try:
+            response = await client.fetch(request)
+            if response.code != 200:
+                LOGGER.error("Failed to evaluate access policy for %s.", input_data)
+                return False
+            response_body = json.loads(response.body.decode())
+            return "result" in response_body and response_body["result"] is True
+        except Exception:
+            LOGGER.exception("Failed to evaluate access policy for %s.", input_data)
+            return False
+
+
+class OpaServer:
+
+    def __init__(self) -> None:
+        self.process: asyncio.subprocess.Process | None = None
+        self.running = False
+
+    async def get_addr_policy_engine(self) -> str:
+        """
+        Returns the "<host>:<port>" on which the policy engine should listen for incoming connections.
+        """
+        return f"{policy_engine_bind_address.get()}:{policy_engine_bind_port.get()}"
+
+    def _initialize_storage(self) -> str:
+        """
+        Make sure the required directories exist in the state directory for the policy engine.
+        """
+        state_dir = config.state_dir.get()
+        policy_engine_state_dir = os.path.join(state_dir, "policy_engine")
+        os.makedirs(policy_engine_state_dir, exist_ok=True)
+        return policy_engine_state_dir
+
+    async def start(self) -> None:
+        if not os.path.isfile(policy_file.get()):
+            raise Exception(f"Access policy file {policy_file.get()} not found.")
+
+        state_dir = self._initialize_storage()
+
+        # Write data to state directory
+        data_file = os.path.join(state_dir, "data.json")
+        data = decorators.AuthorizationMetadata.get_open_policy_agent_data()
+        with open(data_file, "w") as fh:
+            json.dump(data, fh)
+
+        # Start policy engine
+        opa_binary = str(files("inmanta.opa").joinpath("opa").absolute())
+        log_dir = config.log_dir.get()
+        policy_engine_log = os.path.join(log_dir, "policy_engine.log")
+        log_file_handle = None
+        try:
+            log_file_handle = open(policy_engine_log, "wb+")
+            self.process = await asyncio.create_subprocess_exec(
+                opa_binary,
+                "run",
+                "--server",
+                "--addr",
+                await self.get_addr_policy_engine(),
+                "--log-format",
+                "text",
+                "--log-level",
+                "debug",
```

Comment:
do we always want debug? (just asking)

Reply:
I wasn't sure either. But if we don't make this configurable, debug is the only real option. But it's probably better if I add a config option for this.